substrate/frame/examples/tasks/src/lib.rs

```diff
@@ -0,0 +1,77 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::dispatch::DispatchResult;
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+pub mod mock;
+pub mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub mod weights;
+pub use weights::*;
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet {
+
+	use super::*;
+	use frame_support::pallet_prelude::*;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NotFound,
+	}
+
+	#[pallet::tasks]
+	impl<T: Config> Pallet<T> {
+		/// Add a pair of numbers into the totals and remove them.
+		#[pallet::task_list(Numbers::<T>::iter_keys())]
+		#[pallet::task_condition(|i| Numbers::<T>::contains_key(i))]
+		#[pallet::task_weight(T::WeightInfo::add_number_into_total())]
+		#[pallet::task_index(0)]
```

Comment:
This probably deviates too much from the original syntax, but is IMHO a bit more readable.
```suggestion
		#[pallet::task(Task {
				list: Numbers::<T>::iter_keys(),
				condition: &Numbers::<T>::contains_key,
				weight: T::WeightInfo::add_number_into_total(),
				index: 0,
		)]
```

Reply:
It's not so much that it deviates from the original syntax. Rather it deviates from pre-existing idioms, specifically for `pallet::call`:

```rust
		#[pallet::weight(Weight::zero())]
		#[pallet::call_index(0)]
		...
```